### PR TITLE
New algorithms "Extract Z values" and "Extract m values"

### LINKED ISSUE #3295: [FEATURE][processing] New algorithms "Extract Z values" and "Extract m values"
Original commit: https://github.com/qgis/QGIS/commit/cef9c4eaec4ccf33bda7bc6d0cff5a6b8d66e488 by nyalldawson

These algorithms allow users to convert z or m values present in feature
geometries to attributes in the layer. By default the z/m value from the
first vertex in the feature is extracted, but optionally statistics
can be calculated on ALL the z/m values from the geometry (e.g. calculating
mean/min/max/sum/etc of z values).

### PR DESCRIPTION
**Pull request by havatv**
Goal:

Ticket(s): Fixes #3295

- [ ] Backport to LTR documentation is required

### CODE REVIEW

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 1]
```diff
@@ -1531,6 +1531,96 @@ Outputs
...
+     - Name
+     - Type
+     - Description
+   * - **Input layer**
+     - ``INPUT``
+     - [vector: any]
+     - Input vector layer
+   * - **Summaries to calculate**
+     - ``SUMMARIES``
+     - [enumeration]
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]:
```suggestion
     - [enumeration]

       Default: [0]
```
Hoping that i wrote it the right way

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 2]
```diff
@@ -1531,6 +1531,96 @@ Outputs
...
+   :widths: 20 20 20 40
+   :stub-columns: 0
+
+   * - Label
+     - Name
+     - Type
+     - Description
+   * - **Extracted**
+     - ``OUTPUT``
+     - [vector: polygon]
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]:
```suggestion
     - [same as input]
```

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 3]
```diff
@@ -1531,6 +1531,96 @@ Outputs
...
+     - Bounding boxes of input layer
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]:
```suggestion
     - The output vector layer (with M values)
```

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 4]
```diff
@@ -1610,6 +1700,96 @@ Outputs
...
+
+Extract Z values
+----------------
+
+Extracts Z values from geometries into feature attributes.
+
+By default only the Z value from the first vertex of each
+feature is extracted, however the algorithm can optionally
+calculate statistics on all of the geometry's Z values,
+including sums, means, and minimums and maximums.
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]:
```suggestion
including sum, mean, minimum and maximum.
```

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 5]
```diff
@@ -1610,6 +1700,96 @@ Outputs
...
+
+.. seealso:: :ref:`qgisextractmvalues`
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]:
```suggestion
.. seealso:: :ref:`qgisextractzvalues`, :ref:`qgissetzvalue`, :ref:`qgisdropmzvalues`
```
with qgissetzfromraster?

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 6]
```diff
@@ -1610,6 +1700,96 @@ Outputs
...
+     - Name
+     - Type
+     - Description
+   * - **Input layer**
+     - ``INPUT``
+     - [vector: any]
+     - Input vector layer
+   * - **Summaries to calculate**
+     - ``SUMMARIES``
+     - [enumeration]
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]:
```suggestion
     - [enumeration]
     
       Default: [0]
```

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 7]
```diff
@@ -1610,6 +1700,96 @@ Outputs
...
+     - Statistics on the M values of a geometry.
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]:
```suggestion
     - Statistics on the Z values of a geometry.
```

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 8]
```diff
@@ -1610,6 +1700,96 @@ Outputs
...
+       * 12 --- Variety
+       * 13 --- Q1
+       * 14 --- Q3
+       * 15 --- IQR
+
+   * - **Output column prefix**
+     - ``COLUMN_PREFIX``
+     - [string]
+
+       Default: 'm\_'
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]:
```suggestion
       Default: 'z\_'
```

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 9]
```diff
@@ -1610,6 +1700,96 @@ Outputs
...
+     - The prefix for the output (M) column
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]:
```suggestion
     - The prefix for the output (Z) column
```

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 10]
```diff
@@ -1610,6 +1700,96 @@ Outputs
...
+   :widths: 20 20 20 40
+   :stub-columns: 0
+
+   * - Label
+     - Name
+     - Type
+     - Description
+   * - **Extracted**
+     - ``OUTPUT``
+     - [vector: polygon]
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]:
```suggestion
     - [same as input]
```

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 11]
```diff
@@ -1531,6 +1531,96 @@ Outputs
...
+
+Extract M values
+----------------
+
+Extracts M values from geometries into feature attributes.
+
+By default only the M value from the first vertex of each
+feature is extracted, however the algorithm can optionally
+calculate statistics on all of the geometry's M values,
+including sums, means, and minimums and maximums.
```
[the remark this answers is not in the document]
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: I agree! This was just copy and paste from the algorithm help.

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst (hunk 1 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: Perfect. Thanks!

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst (hunk 2 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: Yes!

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst (hunk 3 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: Thanks.

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst (hunk 8 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: Sorry!

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst (hunk 10 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: Thanks!

**Pull request merged**